source/deployment/elasticsearch.rst

```diff
@@ -0,0 +1,73 @@
+
+Configuring Elasticsearch in Mattermost
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Follow these steps to connect your Elasticsearch server to Mattermost and generate the post index.
+
+1. Open the **System Console** > **ADVANCED** > **Elasticsearch (Beta)** section.
+2. Set **Enable Elasticsearch Indexing** to `true` to enable the other the settings on the page. Once the configuration is saved, new posts made to the database will be automatically indexed on the Elasticsearch server.
+3. Set the Elasticsearch server connection details:
+  a) Enter **Server Connection Address**
+  b) Enter **Server Username** used to access the Elasticsearch server (Optional)
+    - Note: For AWS Elasticsearch leave this field blank
+  c) Enter **Server Password** associated with the username (Optional)
+    - Note: For AWS Elasticsearch leave this field blank
+  d) Set **Enable Cluster Sniffing**. Sniffing finds and connects to all data nodes in your cluster automatically.
+    - Note: For AWS Elasticsearch this field should be set to `false`
+4. Click **Test Connection** and **Save** the configuration.
+  - If the server connection is unsuccessful you will not be able to save the configuration or enable searching with Elasticsearch
+5. Build the post index of existing posts by clicking **Build Index**
+  - This process can take up to a few hours depending on the size of the post database and number of messages. The progress percentage can be seen as the index is created.
+6. Enable Elasticsearch by setting **Enable Elasticsearch for search queries** to `true`
+  - Note: It is recommended that bulk indexing be completed before enabling Elasticsearch, otherwise search results will be incomplete.
+7. Restart the Mattermost server
+
+
+Beta Limitations
+-----------------
+
+1. Elasticsearch uses a standard selection of "stop words" that are filtered out of search results: 
+
+  - "a", "an", "and", "are", "as", "at", "be", "but", "by", "for", "if", "in", "into", "is", "it", "no", "not", "of", "on", "or", "such", "that", "the", "their", "then", "there", "these", "they", "this", "to", "was", "will", "with"  
+
+2. Known Issues in Beta:
+
+  - Searching stop words in quotes returns more results than just the searched terms (`ticket <https://mattermost.atlassian.net/browse/PLT-7314>`_)
+  - Pressing ESC in the search box clears the search text (`ticket <https://mattermost.atlassian.net/browse/PLT-7368>`_)
+  - AWS Elasticsearch implementations have a limit of 1000 days of post history that is searchable in Beta.
+  - Highlighting of search terms is sometimes missing from the results list
+
+Frequently Asked Questions (FAQ)
+--------------------------------
+
+Why do I need to use Elasticsearch?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Due to the limitations of scaling with database search, Enterprise deployments with more than 5,000,000 messages require Elasticsearch in order to execute performant search queries.
```

Comment:
Maybe we should check with leads on this. 

Given the Uber article which says Mattermost supports 1 million messages a day, I worry it would appear odd that search only supports 5 million: https://eng.uber.com/uchat/

This is the blob in our 4.1 announce

```
Enterprise Edition E20 now offers a highly available elasticsearch engine in Beta.

A high availability cluster enables a Mattermost system to maintain service during outages and hardware failures through the use of redundant infrastructure. With a new Elasticsearch engine, you can now run highly efficient database searches in a cluster environment. 

Furthermore, you can our search replica feature to isolate search queries onto one or more database servers. A search replica is similar to a read replica, but is used only for handling search queries.

```